compiler/rustc_session/src/config.rs

```diff
@@ -1061,35 +1071,114 @@ pub fn to_crate_check_config(cfg: CheckCfg) -> CrateCheckConfig {
 impl CrateCheckConfig {
     /// Fills a `CrateCheckConfig` with well-known configuration names.
     pub fn fill_well_known(&mut self) {
-        // NOTE: This should be kept in sync with `default_configuration`
-        const WELL_KNOWN_NAMES: &[Symbol] = &[
-            sym::unix,
-            sym::windows,
-            sym::target_os,
-            sym::target_family,
-            sym::target_arch,
-            sym::target_endian,
-            sym::target_pointer_width,
-            sym::target_env,
-            sym::target_abi,
-            sym::target_vendor,
-            sym::target_thread_local,
-            sym::target_has_atomic_load_store,
-            sym::target_has_atomic,
-            sym::target_has_atomic_equal_alignment,
-            sym::panic,
-            sym::sanitize,
-            sym::debug_assertions,
-            sym::proc_macro,
-            sym::test,
-            sym::doc,
-            sym::doctest,
-            sym::feature,
+        enum Values<'a> {
+            None,
+            Unknown,
+            FromSlice(&'a [Symbol]),
+            FromIterator(&'a mut dyn Iterator<Item = Symbol>),
+        }
+
+        let panic_values = &PanicStrategy::all();
+
+        let atomic_values = &[
+            sym::ptr,
+            sym::integer(8usize),
+            sym::integer(16usize),
+            sym::integer(32usize),
+            sym::integer(64usize),
+            sym::integer(128usize),
         ];
-        if let Some(names_valid) = &mut self.names_valid {
-            for &name in WELL_KNOWN_NAMES {
+
+        let mut sanitize_values = SanitizerSet::all()
+            .into_iter()
+            .map(|sanitizer| Symbol::intern(sanitizer.as_str().unwrap()));
+
+        let targets = if self.well_known_values {
+            // FIXME(Urgau): We should get all the targets at compile-time but
+            // this is not yet possible with the current state of const functions and iterators.
```

Comment:
I don't think built-in target loading is possible at compiler's compile time, because it needs to read some environment variables from the host at compiler's runtime (at least on Apple targets, see e.g. `compiler\rustc_target\src\spec\apple_base.rs`).

Reply:
Oh! That's a shame. To bad.